opm/autodiff/SolventPropsAdFromDeck.cpp

```diff
@@ -402,18 +387,57 @@ V SolventPropsAdFromDeck::solventSurfaceDensity(const Cells& cells) const {
     const int n = cells.size();
     V density(n);
     for (int i = 0; i < n; ++i) {
-        int regionIdx = cellPvtRegionIdx_[i];
+        int regionIdx = cellPvtRegionIdx_[cells[i]];
         density[i] = solvent_surface_densities_[regionIdx];
     }
     return density;
 }
 
-double SolventPropsAdFromDeck::mixingParameterViscosity() const {
-    return mix_param_viscosity_;
+V SolventPropsAdFromDeck::mixingParameterViscosity(const Cells& cells) const {
+    const int n = cells.size();
+    if (mix_param_viscosity_.size() > 0) {
+        V mix_param(n);
+        for (int i = 0; i < n; ++i) {
+            int regionIdx = cellMiscRegionIdx_[cells[i]];
+            mix_param[i] = mix_param_viscosity_[regionIdx];
+        }
+        return mix_param;
+    }
+    // return zeros if not specified
+    return V::Zero(n);
+}
+
+V SolventPropsAdFromDeck::mixingParameterDensity(const Cells& cells) const {
+    const int n = cells.size();
+    if (mix_param_viscosity_.size() > 0) {
+        V mix_param(n);
+        for (int i = 0; i < n; ++i) {
+            int regionIdx = cellMiscRegionIdx_[cells[i]];
+            mix_param[i] = mix_param_density_[regionIdx];
+        }
+        return mix_param;
+    }
+    // return zeros if not specified
+    return V::Zero(n);
 }
 
-double SolventPropsAdFromDeck::mixingParameterDensity() const {
-    return mix_param_density_;
+void SolventPropsAdFromDeck::extractTableIndex(const std::string& keyword,
```

Comment:
Strange indentation, also prefer to associate & and * with type not variable name, and put output parameter last.